matlab/CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+cmake_minimum_required(VERSION 3.7.0)
+
+# If Matlab_ROOT_DIR is set, the Matlab at that location is used.
+find_package(Matlab COMPONENTS MX_LIBRARY REQUIRED)
+
+# Dependencies of the modules:
+#check installed version in order to include the correct OpenCV libraries
+#version variable is defined from project root's CMakeLists
+if("${OpenCV_VERSION_MAJOR}$" EQUAL 2)
+  message(STATUS "OpenCV 2.x detected")
+  find_package(OpenCV 2.4.3 REQUIRED core imgproc highgui)
+elseif("${OpenCV_VERSION_MAJOR}$" EQUAL 3)
+  message(STATUS "OpenCV 3.x detected - including imgcodecs for compatibility")
+  find_package(OpenCV 3 REQUIRED core imgproc imgcodecs)
+endif()
+set_target_properties(${OpenCV_LIBS} PROPERTIES MAP_IMPORTED_CONFIG_RELWITHDEBINFO RELEASE)
+
+if(MSVC)
```

Comment:
Modified (copied changes from examples folder).  Caveat: requires matlab for testing.